artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/OpenWireConnection.java

```diff
@@ -430,24 +442,29 @@ private void callFailureListeners(final ActiveMQException me) {
 
    // send a WireFormatInfo to the peer
    public void sendHandshake() {
-      WireFormatInfo info = wireFormat.getPreferedWireFormatInfo();
+      WireFormatInfo info = inWireFormat.getPreferedWireFormatInfo();
       sendCommand(info);
    }
 
    public ConnectionState getState() {
       return state;
    }
 
+   private static void tracePhysicalSend(Connection transportConnection, Command command) {
+      logger.trace("connectionID: " + (transportConnection == null ? "" : transportConnection.getID()) + " SENDING: " + (command == null ? "NULL" : command));
+   }
+
    public void physicalSend(Command command) throws IOException {
 
       if (logger.isTraceEnabled()) {
-         logger.trace("connectionID: " + (getTransportConnection() == null ? "" : getTransportConnection().getID())
-                         + " SENDING: " + (command == null ? "NULL" : command));
+         tracePhysicalSend(transportConnection, command);
       }
 
       try {
-         ByteSequence bytes = wireFormat.marshal(command);
-         ActiveMQBuffer buffer = OpenWireUtil.toActiveMQBuffer(bytes);
+         final ByteSequence bytes = outWireFormat.marshal(command);
+         final int bufferSize = bytes.length;
+         final ActiveMQBuffer buffer = transportConnection.createTransportBuffer(bufferSize);
```

Comment:
should this be getTransportConnection, like the existing code below, or should the existing code below use the field directly also. Just a consistency POV.